core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java

```diff
@@ -88,4 +100,32 @@ public static Type resolveType(Type type, Type contextType) {
     public static boolean isArray(Type type) {
         return type instanceof Class<?> && ((Class<?>) type).isArray();
     }
+
+    /**
+     * Given a subtype of {@code Map<K,V>}, returns the corresponding map entry type {@code Map.Entry<K,V>}.
+     * @param mapType the map subtype
+     * @return the map entry type
+     */
+    public static Type resolveMapEntryType(Type mapType) {
+        Type keyType = resolveType(KEY, mapType);
+        Type valueType = resolveType(VALUE, mapType);
+        return resolveMapEntryType(keyType, valueType);
+    }
+
+    /**
+     * Given a key and value type, returns the map entry type {@code Map.Entry<keyType,valueType>}.
+     * @param keyType the key type
+     * @param valueType the value type
+     * @return the map entry type
+     */
+    public static Type resolveMapEntryType(Type keyType, Type valueType) {
+        return resolveMapEntryType(TypeToken.of(keyType), TypeToken.of(valueType));
+    }
+
+    private static <K, V> Type resolveMapEntryType(TypeToken<K> keyType, TypeToken<V> valueType) {
```

Comment:
Yeah, this TypeToken use case is a little rough around the edges, but it gets the job done.